nanoFramework.CoreLibrary/System/Guid.cs

```diff
@@ -103,7 +108,7 @@ public Guid(byte[] b)
 #pragma warning disable S3928 // Parameter names used into ArgumentException constructors should match an existing one 
             if (b.Length != 16)
             {
-                throw new ArgumentException();
+                throw new ArgumentException("Byte array must be 16 bytes long.", nameof(b));
```

Comment:
The project standard policy is to not use descriptive messages on exceptions.
Reason is that it wastes a lot of flash space. Most of the times, the exception reason is perfectly identifiable and the description goes into the Intellisense comment. 😉